kedro/framework/cli/jupyter.py

```diff
@@ -3,79 +3,37 @@
 """
 import json
 import os
-import re
+import shutil
 import sys
 from collections import Counter
 from glob import iglob
 from pathlib import Path
-from typing import Any, Dict, Iterable, List
+from typing import Any, Dict
 from warnings import warn
 
 import click
 from click import secho
-from jupyter_client.kernelspec import NATIVE_KERNEL_NAME, KernelSpecManager
-from traitlets import Unicode
+from ipykernel.kernelspec import install
+from jupyter_client.kernelspec import find_kernel_specs
 
 from kedro.framework.cli.utils import (
     KedroCliError,
     _check_module_importable,
     command_with_verbosity,
     env_option,
     forward_command,
-    ipython_message,
-    load_entry_points,
     python_call,
 )
 from kedro.framework.project import validate_settings
 from kedro.framework.startup import ProjectMetadata
 
-JUPYTER_IP_HELP = "IP address of the Jupyter server."
-JUPYTER_ALL_KERNELS_HELP = "Display all available Python kernels."
-JUPYTER_IDLE_TIMEOUT_HELP = """When a notebook is closed, Jupyter server will
-terminate its kernel after so many seconds of inactivity. This does not affect
-any open notebooks."""
-
 CONVERT_ALL_HELP = """Extract the nodes from all notebooks in the Kedro project directory,
 including sub-folders."""
 
 OVERWRITE_HELP = """If Python file already exists for the equivalent notebook,
 overwrite its contents."""
 
 
-def collect_line_magic():
-    """Interface function for collecting line magic functions from plugin entry points."""
-    return load_entry_points("line_magic")
```

Comment:
What is this going to be replaced by? How would stuff like `%run_viz` work?

Reply:
This wasn't actually used anywhere and I think has just been overlooked for deletion before. It's done here instead now: https://github.com/kedro-org/kedro/blob/main/kedro/extras/extensions/ipython.py#L85-L87.